sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/AdtQueryBuilder.cs

```diff
@@ -0,0 +1,117 @@
+        }
+
+        /// <summary>
+        /// Specifies the list of all possible columns to return.
+        /// </summary>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectAll()
+        {
+            return _selectQuery.SelectAll();
+        }
+
+        /// <summary>
+        /// Used to return only a certain number of top items that meet the query requirements.
+        /// </summary>
+        /// <param name="count"> The argument for TOP(), i.e. the number of results to return. </param>
+        /// <param name="args"> The arguments that can be optionally passed with top (e.g., property name). </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectTop(int count, params string[] args)
+        {
+            return _selectQuery.SelectTop(count, args);
+        }
+
+        /// <summary>
+        /// Used to return only a certain number of top items that meet the query requirements.
+        /// </summary>
+        /// <param name="count"> The argument for TOP(), i.e. the number of results to return. </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectTopAll(int count)
+        {
+            return _selectQuery.SelectTopAll(count);
+        }
+
+        /// <summary>
+        /// Used to return the number of items that meet the query requirements.
+        /// </summary>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectCount()
+        {
+            return _selectQuery.SelectCount();
+        }
+
+        /// <summary>
+        /// Used to provide the entire SELECT clause as opposed to providing individual field (column) names.
+        /// </summary>
+        /// <example>
+        /// <code snippet="Snippet:DigitalTwinsQueryBuilderOverride" language="csharp">
+        /// // SELECT TOP(3) Room, Temperature FROM DIGITALTWINS
+        /// new AdtQueryBuilder()
+        /// .SelectCustom(&quot;TOP(3) Room, Temperature&quot;)
+        /// .From(AdtCollection.DigitalTwins)
+        /// .Build();
+        /// </code>
+        /// </example>
+        /// <param name="literalQuery"> Query in string format. </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectCustom(string literalQuery)
+        {
+            return _selectQuery.Select(literalQuery);
+        }
+
+        /// <summary>
+        /// Gets the string representation of the built query.
+        /// </summary>
+        /// <returns> String represenation of query. </returns>
+        public string GetQueryText()
+        {
+            StringBuilder finalQuery = new StringBuilder();
+
+            // build the select string
+            finalQuery.Append(_selectQuery.GetQueryText());
+
+            // build the from string
+            finalQuery.Append(' ').Append(_fromQuery.GetQueryText());
```

Comment:
for single spaces, you can alternatively just use `" "` instead of the `' '`.

Reply:
Actually, I'm getting an error suggesting that I use a char instead of a string for appending a single space -- do you know why this might be happening?

Reply:
a character is fine to use